packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -1,470 +1,630 @@
 import { getMultiBridgeIntermediateToken } from './multibridge_utils';
+import { getIntermediateTokens } from './multihop_utils';
 import {
     BalancerFillData,
     BancorFillData,
     BatchedOperation,
     CurveFillData,
     CurveInfo,
     DexSample,
-    FillData,
-    Quote,
+    ERC20BridgeSource,
+    HopInfo,
+    LiquidityProviderFillData,
+    MultiBridgeFillData,
+    MultiHopFillData,
+    SamplerContractOperation,
     SourceQuoteOperation,
+    TokenAdjacencyGraph,
     UniswapV2FillData,
 } from './types';
 
+// tslint:disable:no-inferred-empty-object-type no-unbound-method
+
 /**
  * Composable operations that can be batched in a single transaction,
  * for use with `DexOrderSampler.executeAsync()`.
  */
-export const samplerOperations = {
-    getOrderFillableTakerAmounts(orders: SignedOrder[], exchangeAddress: string): BatchedOperation<BigNumber[]> {
-        return {
-            encodeCall: contract => {
-                return contract
-                    .getOrderFillableTakerAssetAmounts(orders, orders.map(o => o.signature), exchangeAddress)
-                    .getABIEncodedTransactionData();
-            },
-            handleCallResultsAsync: async (contract, callResults) => {
-                return contract.getABIDecodedReturnData<BigNumber[]>('getOrderFillableTakerAssetAmounts', callResults);
-            },
-        };
-    },
-    getOrderFillableMakerAmounts(orders: SignedOrder[], exchangeAddress: string): BatchedOperation<BigNumber[]> {
+export class SamplerOperations {
+    public static constant<T>(result: T): BatchedOperation<T> {
         return {
-            encodeCall: contract => {
-                return contract
-                    .getOrderFillableMakerAssetAmounts(orders, orders.map(o => o.signature), exchangeAddress)
-                    .getABIEncodedTransactionData();
+            encodeCall: () => {
+                return '0x';
             },
-            handleCallResultsAsync: async (contract, callResults) => {
-                return contract.getABIDecodedReturnData<BigNumber[]>('getOrderFillableMakerAssetAmounts', callResults);
+            handleCallResultsAsync: async _callResults => {
+                return result;
             },
         };
-    },
-    getKyberSellQuotes(makerToken: string, takerToken: string, takerFillAmounts: BigNumber[]): SourceQuoteOperation {
-        return {
+    }
+
+    constructor(
+        protected readonly _samplerContract: ERC20BridgeSamplerContract,
+        public readonly bancorService?: BancorService,
+        public readonly balancerPoolsCache: BalancerPoolsCache = new BalancerPoolsCache(),
+    ) {}
+
+    public getOrderFillableTakerAmounts(orders: SignedOrder[], exchangeAddress: string): BatchedOperation<BigNumber[]> {
+        return new SamplerContractOperation({
+            source: ERC20BridgeSource.Native,
+            contract: this._samplerContract,
+            function: this._samplerContract.getOrderFillableTakerAssetAmounts,
```

Comment:
Isn't this annoying since you lose type information?

Reply:
fixed